mailbox/backup/src/main/java/org/apache/james/mailbox/backup/DefaultMailboxBackup.java

```diff
@@ -93,13 +94,28 @@ public void backupAccount(User user, OutputStream destination) throws IOExceptio
         archive(mailboxes, messages, destination);
     }
 
+    private boolean isAccountNonEmpty(User user) throws BadCredentialsException, MailboxException, IOException {
+        MailboxSession session = mailboxManager.createSystemSession(user.asString());
+        return getAccountContentForUser(session).stream().findFirst().isPresent();
+    }
+
     @Override
-    public Publisher<Void> restore(User user, InputStream source) {
+    public Publisher<BackupStatus> restore(User user, InputStream source) {
+        try {
+            if (isAccountNonEmpty(user)) {
+                return Mono.just(BackupStatus.NON_EMPTY_RECEIVER_ACCOUNT);
+            }
+        } catch (Exception e) {
+            LOGGER.error("Error during account restoration for user : " + user, e);
+            return Mono.just(BackupStatus.FAILED);
```

Comment:
Maybe using `Mono.error` and associated mechanisms should be used here.

Reply:
We don't want to expose Mono in this API.